### ISSUE TITLE
[BUG] Native test on Windows failing often on main

### ISSUE THREAD

**Issue by apupier** [KaotoIO/kaoto-backend#412]
**Describe the bug**

main branch is broken

https://github.com/KaotoIO/kaoto-backend/actions/runs/3940485034/jobs/6741659665#step:5:7981

**To Reproduce**

**Expected behavior**

**Logs**

```
Error:  Errors: 
Error:    DeploymentGeneratorServiceTest.parse � Runtime java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
Error: ]: Build step io.quarkus.kubernetes.client.deployment.DevServicesKubernetesProcessor#setupKubernetesDevService threw an exception: java.lang.RuntimeException: java.lang.IllegalStateException: Previous attempts to find a Docker environment failed. Will not retry. Please see logs and check configuration
	at io.quarkus.kubernetes.client.deployment.DevServicesKubernetesProcessor.setupKubernetesDevService(DevServicesKubernetesProcessor.java:119)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:909)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:281)
	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)
	at java.base/java.lang.Thread.run(Thread.java:833)
	at org.jboss.threads.JBossThread.run(JBossThread.java:501)
Caused by: java.lang.IllegalStateException: Previous attempts to find a Docker environment failed. Will not retry. Please see logs and check configuration
	at org.testcontainers.dockerclient.DockerClientProviderStrategy.getFirstValidStrategy(DockerClientProviderStrategy.java:212)
	at org.testcontainers.DockerClientFactory.getOrInitializeStrategy(DockerClientFactory.java:150)
	at org.testcontainers.DockerClientFactory.dockerHostIpAddress(DockerClientFactory.java:323)
	at org.testcontainers.containers.ContainerState.getHost(ContainerState.java:64)
	at com.dajudge.kindcontainer.ApiServerContainer.<init>(ApiServerContainer.java:49)
	at io.quarkus.kubernetes.client.deployment.DevServicesKubernetesProcessor.lambda$startKubernetes$7(DevServicesKubernetesProcessor.java:203)
	at java.base/java.util.Optional.orElseGet(Optional.java:364)
	at io.quarkus.kubernetes.client.deployment.DevServicesKubernetesProcessor.startKubernetes(DevServicesKubernetesProcessor.java:241)
	at io.quarkus.kubernetes.client.deployment.DevServicesKubernetesProcessor.setupKubernetesDevService(DevServicesKubernetesProcessor.java:109)
	... 11 more
```

**Environment (please complete the following information):**
 - OS: [e.g. "Linux 5.10.0-9-amd64 #1 SMP Debian 5.10.70-1 (2021-09-30) x86_64 GNU/Linux"]
 - Output of the mvn -version` command
 - Kaoto Version [e.g. 1.0]

**Additional context**

**Comment by apupier** [comment]
last success on main: https://github.com/KaotoIO/kaoto-backend/actions/runs/3828055804

First failure on main: https://github.com/KaotoIO/kaoto-backend/actions/runs/3829876675 

there was some Pull Request in success since that time, need to check if they were based on top of the commit of first failure or on an older commit

**Comment by apupier** [comment]
switching back to Quarkus 2.14 instead of 2.15 is effectively fixing the build on Windows native: https://github.com/KaotoIO/kaoto-backend/actions/runs/3947008899/jobs/6755520828

**Comment by apupier** [comment]
one success with main: https://github.com/KaotoIO/kaoto-backend/actions/runs/3940485034/jobs/6755545347 so we have less than 1 out of 10 success :-s

**Comment by apupier** [comment]
some trace found:

```
[INFO] Running io.kaoto.backend.api.service.deployment.generator.camelroute.DeploymentGeneratorServiceTest
2023-01-18 08:56:51,815 WARN  [io.fab.kub.cli.Config] (build-26) Error reading service account token from: [/var/run/secrets/kubernetes.io/serviceaccount/token]. Ignoring.
2023-01-18 08:56:53,625 INFO  [io.qua.arc.pro.BeanProcessor] (build-40) Found unrecommended usage of private members (use package-private instead) in application beans:
	- @Inject field io.kaoto.backend.api.service.deployment.generator.camelroute.IntegrationDeploymentGeneratorService#kdgs,
	- Bean constructor void <init>() on io.kaoto.backend.metadata.parser.step.camelroute.CamelRouteParseCatalog,
	- @Inject field io.kaoto.backend.api.service.deployment.generator.kamelet.KameletDeploymentGeneratorService#stepParserService
	- and 2 more - please enable debug logging to see the full list
2023-01-18 08:56:53,876 INFO  [org.tes.doc.DockerClientProviderStrategy] (build-26) Found Docker environment with local Npipe socket (npipe:////./pipe/docker_engine)
2023-01-18 08:56:53,876 WARN  [org.tes.doc.DockerClientProviderStrategy] (build-26) windows is currently not supported
2023-01-18 08:56:53,907 INFO  [org.tes.doc.DockerMachineClientProviderStrategy] (build-26) docker-machine executable was not found on PATH ([C:\Program Files\PowerShell\7, C:\hostedtoolcache\windows\graalvm-ce-java17-windows\22.3.0\x64\graalvm-ce-java17-22.3.0\bin, C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.34.31933\bin\HostX64\x64, C:\Program Files\Microsoft Visual Studio\2022\Enterprise\Common7\IDE\VC\VCPackages, C:\Program Files\Microsoft Visual Studio\2022\Enterprise\Common7\IDE\CommonExtensions\Microsoft\TestWindow, C:\Program Files\Microsoft Visual Studio\2022\Enterprise\Common7\IDE\CommonExtensions\Microsoft\TeamFoundation\Team Explorer, C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Current\bin\Roslyn, C:\Program Files\Microsoft Visual Studio\2022\Enterprise\Team Tools\Performance Tools\x64, C:\Program Files\Microsoft Visual Studio\2022\Enterprise\Team Tools\Performance Tools, C:\Program Files (x86)\Microsoft SDKs\Windows\v10.0A\bin\NETFX 4.8 Tools\x64\, C:\Program Files (x86)\HTML Help Workshop, C:\Program Files\Microsoft Visual Studio\2022\Enterprise\Common7\IDE\CommonExtensions\Microsoft\FSharp\Tools, C:\Program Files\Microsoft Visual Studio\2022\Enterprise\Common7\IDE\Extensions\Microsoft\CodeCoverage.Console, C:\Program Files (x86)\Windows Kits\10\bin\10.0.22621.0\\x64, C:\Program Files (x86)\Windows Kits\10\bin\\x64, C:\Program Files\Microsoft Visual Studio\2022\Enterprise\\MSBuild\Current\Bin\amd64, C:\Windows\Microsoft.NET\Framework64\v4.0.30319, C:\Program Files\Microsoft Visual Studio\2022\Enterprise\Common7\IDE\, C:\Program Files\Microsoft Visual Studio\2022\Enterprise\Common7\Tools\, C:\Program Files\MongoDB\Server\5.0\bin, C:\aliyun-cli, C:\vcpkg, C:\Program Files (x86)\NSIS\, C:\tools\zstd, C:\Program Files\Mercurial\, C:\hostedtoolcache\windows\stack\2.9.3\x64, C:\cabal\bin, C:\\ghcup\bin, C:\tools\ghc-9.4.4\bin, C:\Program Files\dotnet, C:\mysql\bin, C:\Program Files\R\R-4.2.2\bin\x64, C:\SeleniumWebDrivers\GeckoDriver, C:\Program Files (x86)\sbt\bin, C:\Program Files (x86)\GitHub CLI, C:\Program Files\Git\bin, C:\Program Files (x86)\pipx_bin, C:\npm\prefix, C:\hostedtoolcache\windows\go\1.17.13\x64\bin, C:\hostedtoolcache\windows\Python\3.9.13\x64\Scripts, C:\hostedtoolcache\windows\Python\3.9.13\x64, C:\hostedtoolcache\windows\Ruby\3.0.5\x64\bin, C:\tools\kotlinc\bin, C:\hostedtoolcache\windows\Java_Temurin-Hotspot_jdk\8.0.352-8\x64\bin, C:\Program Files\ImageMagick-7.1.0-Q16-HDRI, C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\wbin, C:\ProgramData\kind, C:\Program Files\Microsoft\jdk-11.0.12.7-hotspot\bin, C:\Windows\system32, C:\Windows, C:\Windows\System32\Wbem, C:\Windows\System32\WindowsPowerShell\v1.0\, C:\Windows\System32\OpenSSH\, C:\Program Files\dotnet\, C:\ProgramData\Chocolatey\bin, C:\Program Files\PowerShell\7\, C:\Program Files\Microsoft\Web Platform Installer\, C:\Program Files\Microsoft SQL Server\Client SDK\ODBC\170\Tools\Binn\, C:\Program Files\Microsoft SQL Server\150\Tools\Binn\, C:\Program Files\OpenSSL\bin, C:\Strawberry\c\bin, C:\Strawberry\perl\site\bin, C:\Strawberry\perl\bin, C:\ProgramData\chocolatey\lib\pulumi\tools\Pulumi\bin, C:\Program Files\TortoiseSVN\bin, C:\Program Files\CMake\bin, C:\ProgramData\chocolatey\lib\maven\apache-maven-3.8.7\bin, C:\Program Files\Microsoft Service Fabric\bin\Fabric\Fabric.Code, C:\Program Files\Microsoft SDKs\Service Fabric\Tools\ServiceFabricLocalClusterManager, C:\Program Files\nodejs\, C:\Program Files\Git\cmd, C:\Program Files\Git\mingw64\bin, C:\Program Files\Git\usr\bin, C:\Program Files\GitHub CLI\, c:\tools\php, C:\Program Files (x86)\sbt\bin, C:\SeleniumWebDrivers\ChromeDriver\, C:\SeleniumWebDrivers\EdgeDriver\, C:\Program Files\Amazon\AWSCLIV2\, C:\Program Files\Amazon\SessionManagerPlugin\bin\, C:\Program Files\Amazon\AWSSAMCLI\bin\, C:\Program Files\Microsoft SQL Server\130\Tools\Binn\, C:\Program Files\LLVM\bin, C:\Users\runneradmin\.dotnet\tools, C:\Users\runneradmin\.cargo\bin, C:\Users\runneradmin\AppData\Local\Microsoft\WindowsApps, C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\Llvm\x64\bin, C:\Program Files\Microsoft Visual Studio\2022\Enterprise\Common7\IDE\CommonExtensions\Microsoft\CMake\CMake\bin, C:\Program Files\Microsoft Visual Studio\2022\Enterprise\Common7\IDE\CommonExtensions\Microsoft\CMake\Ninja, C:\Program Files\Microsoft Visual Studio\2022\Enterprise\Common7\IDE\VC\Linux\bin\ConnectionManagerExe])
2023-01-18 08:56:53,907 ERROR [org.tes.doc.DockerClientProviderStrategy] (build-26) Could not find a valid Docker environment. Please check configuration. Attempted configurations were:
	NpipeSocketClientProviderStrategy: failed with exception InvalidConfigurationException (windows containers are currently not supported)As no valid configuration was found, execution cannot continue.
See https://www.testcontainers.org/on_failure.html for more details.
```

**Comment by apupier** [comment]
when on success, there is quite similar log:
```

**Comment by apupier** [comment]
asked for help to investigate on Quarkus user chat: https://quarkusio.zulipchat.com/#narrow/stream/187030-users/topic/2.2E14.20to.202.2E15.3A.20Intermittent.20Failure.20on.20GitHub.20Actions.20Windows/near/322000353

**Comment by apupier** [comment]
after increasing PageSize, it had the side effect to fix also this problem. At leat, it wasn't reproduced in last 10 builds (for which previously we had 9 failures, so definitely better)

**Comment by mkralik3** [comment]
This issue is caused by `kubernetes-client.devservices` which comes from `quarkus-kubernetes-client` and it is enabled by default. It happens in tests in project module where kubernetes server is not used during testing (but the module needs `quarkus-kubernetes-client` dependency due to model definitions).

With Quarkus3, it started to be reproducible. The fix is disable `quarkus.kubernetes-client.devservices.enabled` in modules which has `quarkus-kubernetes-client` dependency (e.g. required due to model definitions) but don't use `quarkus-test-kubernetes-client` `@WithKubernetesTestServer` in tests. For Kaoto-backend, it will be a [part](https://github.com/KaotoIO/kaoto-backend/pull/695/commits/51a312eff054069a8e94832ad4cec24dd4b2fb09) of Quarkus3 pull request.


Btw, I created [a reproducer.](https://github.com/mkralik3/quarkus-simple/tree/main) The issue is only on windows platform ( github actions ) and only with two (and more) modules project. I think the problem is due to uncatched exception from testcontainers package which is thrown during test phase of second package. In first module, it is not able to correctly find a docker, so it sets FAIL_FAST_ALWAYS variable to [true](https://github.com/testcontainers/testcontainers-java/blob/main/core/src/main/java/org/testcontainers/dockerclient/DockerClientProviderStrategy.java#L275) and in the second package, since FAIL_FAST_ALWAYS is true, it throws an [IllegalArgumentException](https://github.com/testcontainers/testcontainers-java/blob/main/core/src/main/java/org/testcontainers/dockerclient/DockerClientProviderStrategy.java#L231) which is not catch by Quarkus anymore (I think it was catch before (Quarkus v 2.15) somewhere because that problem was happening irregularly).


TLDR: This will be fixed as part of Quarkus3 update pull request and this issue can be closed. I also need to report all my findings described above into Quarkus team.

**Comment by lordrip** [comment]
Great investigation job @mkralik3 :clap: , thanks for checking it out
